lib/calculator_page.dart

```diff
@@ -10,147 +11,94 @@ class CalculatorPage extends StatefulWidget {
-                      style: const TextStyle(fontSize: 20),
+     CalculatorProvider calculate =
+      Provider.of<CalculatorProvider>(context);
+    return Consumer<CalculatorProvider>(builder: (context, value, child) {
+      return Scaffold(
+        backgroundColor: Colors.deepPurple[100],
+        body: Column(
+          children: [
+            Expanded(
+              child: Container(
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                  children: [
+                    const SizedBox(
+                      height: 50,
                     ),
-                  ),
-                  Container(
+                    Container(
                       padding: const EdgeInsets.all(20),
-                      alignment: Alignment.centerRight,
+                      alignment: Alignment.centerLeft,
                       child: Text(
-                        userAnswer,
+                        calculate.userQuestion,
                         style: const TextStyle(fontSize: 20),
-                      )),
-                ],
+                      ),
+                    ),
+                    Container(
+                        padding: const EdgeInsets.all(20),
+                        alignment: Alignment.centerRight,
+                        child: Text(
+                          calculate.userAnswer,
+                          style: const TextStyle(fontSize: 20),
+                        )),
+                  ],
+                ),
               ),
             ),
-          ),
-          Expanded(
-            flex: 2,
-            child: Container(
-              child: GridView.builder(
-                itemCount: buttons.length,
-                gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
-                  crossAxisCount: 4,
+            Expanded(
+              flex: 2,
+              child: Container(
+                child: GridView.builder(
+                  itemCount: calculate.buttons.length,
+                  gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+                    crossAxisCount: 4,
+                  ),
+                  itemBuilder: (BuildContext context, int index) {
+                    if (index == 0) {
+                      return MyButton(
+                        buttonTapped: () => calculate.eraseAll(),
+                        color: Colors.green,
+                        textColor: Colors.white,
+                        buttonText: calculate.buttons[index],
+                      );
+                    } else if (index == 1) {
+                      return MyButton(
+                        buttonTapped: () => calculate.deleteButton(),
+                        color: Colors.red,
+                        textColor: Colors.white,
+                        buttonText: calculate.buttons[index],
+                      );
+                    } else if (index == calculate.buttons.length - 1) {
```

Comment:
for switch the case will be 

case var lastIndex when index == calculate.buttons.length - 1:

Reply:
done